scripts/screeps_rl_experiment_card.py

```
@@ -273,6 +621,128 @@ def validate_strategy_id(value: str, label: str) -> None:
+    candidate_ids: set[str] = set()
+    for index, candidate in enumerate(candidates):
+        validate_policy_gradient_candidate(candidate, index)
+        assert isinstance(candidate, dict)
+        candidate_policy_id = candidate["candidatePolicyId"]
+        if candidate_policy_id in candidate_ids:
+            raise CardValidationError(f"duplicate policy_gradient candidatePolicyId: {candidate_policy_id}")
+        candidate_ids.add(candidate_policy_id)
+
+    support = first_present(raw, ("runner_support", "runnerSupport"))
+    if not isinstance(support, dict):
+        raise CardValidationError("policy_gradient.runner_support must be a JSON object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)